pkg/unikontainers/ipc_test.go

```diff
@@ -0,0 +1,181 @@
+	nonExistingSockAddr := "/tmp/non_existing.sock"
+
+	// Create a temporary socket file
+	f, err := os.Create(existingSockAddr)
+	if err != nil {
+		t.Fatalf("Failed to create temporary socket file: %v", err)
+	}
+	defer os.Remove(existingSockAddr)
+	f.Close()
+
+	assert.True(t, SockAddrExists(existingSockAddr), "Expected socket address to exist")
+	assert.False(t, SockAddrExists(nonExistingSockAddr), "Expected socket address to not exist")
+}
+
+func TestSendIPCMessage(t *testing.T) {
+	socketAddress := "/tmp/test.sock"
+	message := ReexecStarted
+
+	listener, err := net.Listen("unix", socketAddress)
+	if err != nil {
+		t.Fatalf("Failed to create listener: %v", err)
+	}
+	defer os.Remove(socketAddress)
+	defer listener.Close()
+
+	go func() {
+		conn, err := listener.Accept()
+		if err != nil {
+			t.Errorf("Failed to accept connection: %v", err)
+		}
+		defer conn.Close()
+
+		buf := make([]byte, len(message))
+		n, err := conn.Read(buf)
+		if err != nil {
+			t.Errorf("Failed to read message: %v", err)
+		}
+
+		assert.Equal(t, string(message), string(buf[:n]), "Expected %s, but got %s", message, string(buf[:n]))
+	}()
+
+	time.Sleep(10 * time.Millisecond) // Ensure listener is ready
+
+	err = SendIPCMessage(socketAddress, message)
+	assert.NoError(t, err, "Expected no error in sending IPC message")
+}
+
+func TestSendIPCMessageWithRetry(t *testing.T) {
+	socketAddress := "/tmp/test_retry.sock"
+	message := ReexecStarted
+
+	listener, err := net.Listen("unix", socketAddress)
+	if err != nil {
+		t.Fatalf("Failed to create listener: %v", err)
+	}
+	defer os.Remove(socketAddress)
+	defer listener.Close()
+
+	go func() {
+		conn, err := listener.Accept()
+		if err != nil {
+			t.Errorf("Failed to accept connection: %v", err)
+		}
+		defer conn.Close()
+
+		buf := make([]byte, len(message))
+		n, err := conn.Read(buf)
+		if err != nil {
+			t.Errorf("Failed to read message: %v", err)
+		}
+
+		assert.Equal(t, string(message), string(buf[:n]), "Expected %s, but got %s", message, string(buf[:n]))
+	}()
+
+	time.Sleep(10 * time.Millisecond) // Ensure listener is ready
+
+	err = sendIPCMessageWithRetry(socketAddress, message, true)
+	assert.NoError(t, err, "Expected no error in sending IPC message with retry")
+}
+
+func TestCreateListener(t *testing.T) {
+	socketAddress := "/tmp/test_create_listener.sock"
+
+	listener, err := CreateListener(socketAddress, true)
+	if err != nil {
+		t.Fatalf("Failed to create listener: %v", err)
+	}
+	defer os.Remove(socketAddress)
+	defer listener.Close()
+
+	assert.NotNil(t, listener, "Expected listener to be created")
+}
+
+func TestAwaitMessage(t *testing.T) {
+	socketAddress := "/tmp/test_await_message.sock"
+	expectedMessage := ReexecStarted
+
+	listener, err := CreateListener(socketAddress, true)
+	if err != nil {
+		t.Fatalf("Failed to create listener: %v", err)
+	}
+	defer os.Remove(socketAddress)
+	defer listener.Close()
+
+	go func() {
+		time.Sleep(10 * time.Millisecond) // Ensure listener is ready
```

Comment:
Maybe this one is not needed, since the socket should have been created before the go func.

Reply:
Yes, that seems to be unnecessary

Reply:
Yes, you are correct, this is not necessary. Removed.